lucille-core/src/main/java/com/kmwllc/lucille/core/Document.java

```diff
@@ -257,13 +257,13 @@ static Document createFromJson(String json, UnaryOperator<String> idUpdater)
     return JsonDocument.fromJsonString(json, idUpdater);
   }
 
-  static void validateNotReservedField(String... names) throws IllegalArgumentException {
+  static void validateFieldNames(String... names) throws IllegalArgumentException {
```

Comment:
I think we should make this method "default" rather than "static". That way 1) we don't have to statically import it from each document implementation, 2) document implementations would be able to override it if necessary.